gasprice/feehistory.go

```diff
@@ -0,0 +1,160 @@
+package gasprice
+
+import (
+	"encoding/binary"
+	"errors"
+	"math"
+	"math/big"
+	"sort"
+)
+
+var (
+	ErrInvalidPercentile = errors.New("invalid percentile")
+	ErrBlockCount        = errors.New("blockCount must be greater than 0")
+	ErrBlockNotFound     = errors.New("could not find block")
+)
+
+const (
+	maxBlockRequest = 1024
+)
+
+type cacheKey struct {
+	number      uint64
+	percentiles string
+}
+
+// processedFees contains the results of a processed block.
+type processedFees struct {
+	reward       []uint64
+	baseFee      uint64
+	gasUsedRatio float64
+}
+
+type txGasAndReward struct {
+	gasUsed *big.Int
+	reward  *big.Int
+}
+
+func (g *GasHelper) FeeHistory(blockCount uint64, newestBlock uint64, rewardPercentiles []float64) (
+	*uint64, *[]uint64, *[]float64, *[][]uint64, error) {
```

Comment:
made mistake with the pointers :sweat_smile: 
I followed the geth spec where values are returned unpacked from `gasPrice.FeeHistory`
If you think returning `feeHistoryResult` from the package is better I can make that change :smile:

Reply:
> If you think returning feeHistoryResult from the package is better I can make that change smile

Yep, I do think it is better that way because the caller is more aware of what each return value denotes (since it is encapsulated within the named field). It is especially convenient when there are more than 2-3 return values. Do you agree?